### DISCUSSION TITLE
How to open file from compiler output from the terminal emulator

### DISCUSSION THREAD

**Discussion by IceSentry** [neovim/neovim#30740] [Q&A]
I'm trying to switch from vscode to neovim. A way I like to work in vscode is to use the terminal emulator to run compiler commands. Most compiler will then give a link to a file with a collumn and line number. In vscode I could easily ctrl-click the link and it would open the file with the cursor at the specified location in the editor while keeping the terminal opened and visible. I like this because I can use it with any language and without an LSP. LSPs are nice but sometimes they are very slow and part of the reason I want to switch to neovim is that I don't want to rely on an LSP for everything.

I'm trying to replicate this in neovim but I can't figure out how to do it. I found the `gf` keybind which does part of what I want, but the issue is that it opens the buffer on top of the terminal emulator. It also doesn't seem to respect the line/column number. What I'd like to happen is that it opens the buffer inside the last active pane so the terminal emulator pane stays visible. I don't actually mind if it ends up closing the terminal pane, I just want it to go to the "main" (or at least last active pane if there's more than one) pane instead of opening inside the terminal emulator. 

I don't mind having to write a bit of lua if that's necessary, I just want some kind of direction on how to do this.

I'm trying to read the manual to see how to do that but I don't know what I'm looking for which makes it a lot harder.

**Comment by zeertzjq** [comment]
There is also `:h CTRL-W_gF` that opens it in a separate tabpage.

**Comment by IceSentry** [comment]
So I ended up with this which kinda does what I want. I have no idea if that's the best or most idiomatic way to do this. I used chatgpt to guide me and it kept telling me to use things that don't exist so I suspect there are better ways to do this.

```lua
function open_file_in_last_pane()
  -- Get the current line
  local line = vim.fn.getline '.'

  -- Iterate over each word in the line
  for word in line:gmatch '%S+' do
    -- Check if the word matches the pattern "filename:line:column"
    local filename, lnum, col = word:match '([^:]+):(%d+):(%d+)'

    if filename and lnum and col then
      vim.cmd 'wincmd p' -- open last active window
      vim.cmd('edit ' .. filename)
      vim.fn.cursor(tonumber(lnum), tonumber(col))
      return
    end
  end

  print 'No valid file link found in the current line.'
end

vim.api.nvim_set_keymap('n', '<leader>gf', ':lua open_file_in_last_pane()<CR>', { noremap = true, silent = true })
```
